include/swift/Reflection/TypeLowering.h

```diff
@@ -206,6 +239,16 @@ class ReferenceTypeInfo : public TypeInfo {
     return Refcounting;
   }
 
+  bool readExtraInhabitantIndex(remote::MemoryReader &reader,
+                                remote::RemoteAddress address,
+                                int *extraInhabitantIndex) const {
+    if (getNumExtraInhabitants() == 0) {
+      *extraInhabitantIndex = -1;
+      return true;
+    }
+    return reader.readHeapObjectExtraInhabitantIndex(address, extraInhabitantIndex);
```

Comment:
I suspect `ReferenceTypeInfo` is used for both function pointers and heap pointers, but I'm not quite sure how to distinguish those cases.